### DISCUSSION TITLE
Allow disabling pause cancelling

### DISCUSSION THREAD

**Discussion by SonoSooS** [ppy/osu#14077] [Ideas] [answered]
(migrated from #14075)

If you have your keybindings setup in a way so that an osu!standard keybinding is near the pause keybinding, it's almost impossible to pause without it being constantly cancelled by accident if you accidentally keep hitting the keybinding because it's too close.

This is a good thing in most cases, as to prevent "troll restart" possible in the proprietary client, where pressing pause by accident, following a standard key, having the cursor on top of restart by default is the recipe for a disaster, especially if you're past 97%, and/or if you're in "godmode", causing a possibly good run to be lost due to a timing oversight.

But in my case, having pause binded to NUMPAD_PLUS, and my standard buttons being arrow LEFT and arrow RIGHT, they are mirrored to NUMPAD_4 and NUMPAD_6 respectively.
This is a problem when I try to pause with NUMPAD_PLUS, as I always keep hitting NUMPAD_6 by accident, constantly cancelling the pause.

This issue poses a problem from both sides though:
- adding an option could make this "overpowered" with "pausing is cheating" implemented, because it would make it easier to pause
- not adding an option very annoying for those who always keep running into this problem, especially that hitting pause by accident has no side effect anyway due to the pause timer alone cancelling the mis-press in most cases

**Comment by frenzibyte** [comment]
There's a couple of statements in this post that don't sound right with how the game functions:
 - osu!(lazer) is not aware of this "troll restart" you mentioned, you can only cause that by doing it yourself.
 - one key binding never cancels another, not in the player area at least. if that is not the case for you, then you have an issue likely at your end, please confirm that with keyboard testing sites.
 - you can easily pause however you want whenever you want if you have the pause key next to the osu! action keys, what you can't do however is spamming pauses, there's a 1 second cooldown after each.

With that out of the way, what is the option you're proposing to resolve this "issue" exactly? can you elaborate further on that?

---

Reading the title again, were you proposing to disable the 1 second pause cooldown so that accidental pauses don't affect actual intended pauses? if so, that's your issue to resolve, not the game's.

**Reply by SonoSooS** [reply]
No, it seems like you understood exactly the opposite of what I wanted.

- "troll restart" is only possible in the proprietary client, and it is indeed triggered by the user (pressing pause and immediately a standard key)
- it does though, while playing, holding the pause key and pressing a standard keybinding (in my case X/C and LEFT/RIGHT) it does cancel pausing
 EDIT: my keyboard works properly, and there is not a weird key intercross caused by pressing certain keys at once
- I know

Nope, I know that "pausing is cheating" is implemented and never planned to be removed. I'm just proposing a switch to disable pause cancelling when pressing a standard key while holding pause.

**Reply by frenzibyte** [reply]
Keep in mind these are all proprietary clients. I'm assuming you're referring to osu!(stable) when you mention so.

> - it does though, while playing, holding the pause key and pressing a standard keybinding (in my case X/C and LEFT/RIGHT) it does cancel pausing
EDIT: my keyboard works properly, and there is not a weird key intercross caused by pressing certain keys at once
>
> I'm just proposing a switch to disable pause cancelling when pressing a standard key while holding pause.

This "pause cancelling" you're proposing to disable was never a feature, it's an issue you're currently having if I understand you correctly.

https://user-images.githubusercontent.com/22781491/127734433-34f336ea-7584-43be-91ce-f6da5f5cb4c9.mp4

~~Can you state your platform and record a video while at it (with the keys overlay opened), it's likely to be platform-specific if it's not an issue with your keyboard.~~

It appears that is the case with the keys you mentioned in https://github.com/ppy/osu/issues/14075#issuecomment-890313789, I'll investigate further there.

**Reply by SonoSooS** [reply]
Yeah.

While I did test this issue extensively, I was accidentally hyperfixated on the specific keycombo.
After adjusting my settings, I discovered that it was something else all along, and actually a bug.

**Comment by frenzibyte** [comment, marked as answer]
After investigating further from the above and https://github.com/ppy/osu/issues/14075#issuecomment-890313789, it turns out that this is not an issue, but a conflict in the key bindings you have.

If one of the osu! actions conflict with a global action (not sure about the modifiers case, I don't think you should play like that), then pressing it while holding pause will release the pause (`GlobalActionContainer` has simultanoues binding mode set to `None`).

I believe what needs to be done is a clarification at the bottom of the offended key binding section that a key conflicting with another in another section can potentially cause issues during activation, as I don't think this is solvable or there's an issue to begin with (consider the case where the conflicted global action activates an overlay for example).
